src/styles/_catalog.scss

```diff
@@ -1,10 +1,27 @@
 .catalog {
-  display: flex;
+  display: grid;
   flex-wrap: wrap;
 
   max-width: 944px;
   margin: 0 auto;
 
+  @media screen and (max-width: 488px) {
+    grid-template-columns: repeat(1, 1fr);
+  }
+  
+  @media screen and (min-width: 488px) {
+    grid-template-columns: repeat(2, 1fr);
+  }
+
+  @media screen and (min-width: 768px) {
+    grid-template-columns: repeat(3, 1fr);
+  }
+
+  @media screen and (min-width: 1024px) {
+    grid-template-columns: repeat(4, 1fr);
+  }
+
+
```

Comment:
I suggest a slight improvement. You need to eliminate card margins and use the full power of grid positioning. Use gap property. Also, try to center cards via max-wight which can be changed in media queries. Of course, there are more solutions, but I found that pretty simple)

Reply:
I mean that grid-template-columns can be used once with repeat(auto-fill, min-max()) property